R/fetch_data.R

```diff
@@ -16,93 +16,69 @@
+  glue::glue("http://www.football-data.co.uk/mmz4281/{season_code(season)}/{country_lookup[tolower(country)]}{division}.csv")
 }
 
 #' Get the football-data.co.uk string for a given season
 #' @keywords internal
 season_code <- function(start_year) {
   end_year <- start_year + 1
-  glue::glue("{start_year %% 100}{end_year %% 100}")
+  glue::glue("{sprintf('%02d',start_year %% 100)}{sprintf('%02d',end_year %% 100)}")
 }
 
-#' Column specification for football-data.co.uk csv files
-#' @keywords internal
-col_spec <- readr::cols(
-  .default = readr::col_double(),
-  Div      = readr::col_character(),
-  Date     = readr::col_character(),
-  HomeTeam = readr::col_character(),
-  AwayTeam = readr::col_character(),
-  FTHG     = readr::col_integer(),
-  FTAG     = readr::col_integer(),
-  FTR      = readr::col_character(),
-  HTHG     = readr::col_integer(),
-  HTAG     = readr::col_integer(),
-  HTR      = readr::col_character(),
-  Referee  = readr::col_character(),
-  HS       = readr::col_integer(),
-  AS       = readr::col_integer(),
-  HST      = readr::col_integer(),
-  AST      = readr::col_integer(),
-  HF       = readr::col_integer(),
-  AF       = readr::col_integer(),
-  HC       = readr::col_integer(),
-  AC       = readr::col_integer(),
-  HY       = readr::col_integer(),
-  AY       = readr::col_integer(),
-  HR       = readr::col_integer(),
-  AR       = readr::col_integer()
-)
-
 # TODO: better naming convention for home and away columns - should be easy to use with
 #       tidyr::gather and tidyr::separate
 #' Map for selecting and renaming foorball-data.co.uk columns
 #' @keywords internal
-colname_map <- list(
-  competition     = "Div",
-  date            = "Date",
-  home            = "HomeTeam",
-  away            = "AwayTeam",
-  hgoal           = "FTHG",
-  agoal           = "FTAG",
-  result          = "FTR",
-  hgoal_ht        = "HTHG",
-  agoal_ht        = "HTAG",
-  result_ht       = "HTR",
-  referee         = "Referee",
-  hshot           = "HS",
-  ashot           = "AS",
-  hshot_on_target = "HST",
-  ashot_on_target = "AST",
-  hfoul           = "HF",
-  afoul           = "AF",
-  hcorner         = "HC",
-  acorner         = "AC",
-  hyellow         = "HY",
-  ayellow         = "AY",
-  hred            = "HR",
-  ared            = "AR",
-  closing_h       = "PSCH",
-  closing_d       = "PSCD",
-  closing_a       = "PSCA"
-)
+colname_map <- data.table::data.table(cols = c("Div", "Date", "HomeTeam", "AwayTeam", "FTHG", "FTAG", "FTR", "HTHG", "HTAG",
+"HTR", "Referee", "HS", "AS", "HST", "AST", "HF", "AF", "HC", "AC", "HY", "AY", "HR", "AR"),
+names = c("competition", "date", "home", "away", "hgoal", "agoal", "result", "hgoal_ht", "agoal_ht", "result_ht",
+          "referee", "hshot", "ashot", "hshot_on_target", "ashot_on_target", "hfoul", "afoul", "hcorner",
+          "acorner", "hyellow", "ayellow", "hred", "ared"
+))
```

Comment:
I switched to a dataframe over the list because of the `tidyselect::any_of` function in line 53, which I couldn't get working with the list. Use of data.table here over tibble (and generally over tidyverse) is just a personal preference, your logic for removing the dependency on data.table is sensible.